main.go

```diff
@@ -1,167 +1,97 @@
+			Value:     &config.SlackUsername,
+		},
+		{
+			Path:      "icon-url",
+			Env:       "SENSU_SLACK_ICON_URL",
+			Argument:  "icon-url",
+			Shorthand: "i",
+			Default:   "http://s3-us-west-2.amazonaws.com/sensuapp.org/sensu.png",
+			Usage:     "A URL to an image to use as the user avatar",
+			Value:     &config.SlackIconUrl,
+		},
 	}
 )
 
 func main() {
-	rootCmd := configureRootCommand()
-	if err := rootCmd.Execute(); err != nil {
-		log.Fatal(err)
+	goHandler, _ := sensu.NewGoHandler(&config.PluginConfig, slackConfigOptions, checkArgs, executeHandler)
+	err := goHandler.Execute()
+	if err != nil {
+		fmt.Printf("Error executing plugin: %s", err)
 	}
 }
 
-func configureRootCommand() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "sensu-slack-handler",
-		Short: "The Sensu Go Slack handler for notifying a channel",
-		RunE:  run,
+func checkArgs(_ *types.Event) error {
+	if len(config.SlackWebhookUrl) == 0 {
+		return fmt.Errorf("--webhook-url or SENSU_SLACK_WEHBOOK_URL environment variable is required")
 	}
 
-	/*
-		Sensitive flags
-		default to using envvar value
-		do not mark as required
-		manually test for empty value
-	*/
-	cmd.Flags().StringVarP(&config.SlackWebhookUrl.Value,
-		"webhook-url",
-		"w",
-		os.Getenv("SLACK_WEBHOOK_URL"),
-		"The webhook url to send messages to, defaults to value of SLACK_WEBHOOK_URL env variable")
-
-	cmd.Flags().StringVarP(&config.SlackChannel.Value,
-		"channel",
-		"c",
-		"#general",
-		"The channel to post messages to")
-
-	cmd.Flags().StringVarP(&config.SlackUsername.Value,
-		"username",
-		"u",
-		"sensu",
-		"The username that messages will be sent as")
-
-	cmd.Flags().StringVarP(&config.SlackIconUrl.Value,
-		"icon-url",
-		"i",
-		"http://s3-us-west-2.amazonaws.com/sensuapp.org/sensu.png",
-		"A URL to an image to use as the user avatar")
-
-	cmd.Flags().IntVarP(&config.Timeout,
-		"timeout",
-		"t",
-		10,
-		"The amount of seconds to wait before terminating the handler")
-
-	return cmd
+	return nil
 }
 
-func run(cmd *cobra.Command, args []string) error {
-	if len(args) != 0 {
-		_ = cmd.Help()
-		return errors.New("invalid argument(s) received")
-	}
-
-	// load & parse stdin
-	if stdin == nil {
-		stdin = os.Stdin
-	}
-	eventJSON, err := ioutil.ReadAll(stdin)
-	if err != nil {
-		return fmt.Errorf("failed to read stdin: %s", err.Error())
-	}
-	event := &types.Event{}
-	err = json.Unmarshal(eventJSON, event)
-	if err != nil {
-		return fmt.Errorf("failed to unmarshal stdin data: %s", eventJSON)
-	}
-
-	// configuration validation & overrides
-	if config.SlackWebhookUrl.Value == "" {
-		_ = cmd.Help()
-		return fmt.Errorf("webhook url is empty")
-	}
-
-	configurationOverrides(&config, options, event)
-
-	if err = validateEvent(event); err != nil {
-		return errors.New(err.Error())
-	}
-
-	if err = sendMessage(event); err != nil {
+func executeHandler(event *types.Event) error {
+	if err := sendMessage(event); err != nil {
```

Comment:
This entire function could probably be simplified to `return sendMessage(event)`.. which makes me wonder why the `executeHandler` function exists anymore.